src/tests/resistance/handshake.rs

```diff
@@ -101,6 +111,137 @@ async fn r001_t2_HANDSHAKE_reject_if_server_too_long() {
+    let mut synthetic_node = SyntheticNode::new(&cfg).await;
+
+    // Connect to the node and initiate the handshake.
+    let handshake_established = if synthetic_node.connect(node.addr()).await.is_err() {
+        false
+    } else {
+        // Wait for any message.
+        synthetic_node
+            .expect_message(&|m: &BinaryMessage| matches!(&m, _))
+            .await
+    };
+
+    if debug.is_on() {
+        // Let us see a few more logs from the node before shutdown.
+        sleep(Duration::from_millis(200)).await;
+    }
+
+    // Gracefully shut down the nodes.
+    synthetic_node.shut_down().await;
+    assert_eq!(node.stop().expect(ERR_NODE_STOP), ChildExitCode::Success);
+
+    handshake_established
+}
+
+#[tokio::test]
+#[ignore = "internal test"]
+async fn normal_handshake() {
+    let debug = Debug::enable();
+
+    // Basically, a copy of the C001 test.
+    assert!(
+        run_handshake_req_test_with_cfg(Default::default(), debug).await,
+        "a default configuration doesn't work"
+    );
+}
+
+/// Generate a string with a given length.
+fn gen_huge_string(len: usize) -> String {
+    vec!['y'; len].into_iter().collect::<String>()
+}
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_t3_HANDSHAKE_connection_field() {
+    // ZG-RESISTANCE-001
+    // Expected valid value for the "Connection" field in the handshake should be "Upgrade".
+
+    let debug = Debug::disable();
+
+    let gen_cfg = |connection: String| SynthNodeCfg {
+        handshake: Some(HandshakeCfg {
+            http_connection: connection,
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    // Valid scenarios:
+
+    // These are also valid, but should they be?
+    let cfg = gen_cfg("upgrade".to_owned());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+    let cfg = gen_cfg("uPgRAdE".to_owned());
+    assert!(run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Below tests assert the connection shouldn't be established.
+
+    // Field is almost correct.
+    let cfg = gen_cfg("Upgrad".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+    let cfg = gen_cfg("Upgradee".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+    let cfg = gen_cfg("UpgradeUpgrade".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Find the largest instance value which the node could accept, but won't due to invalid value
+    // in the field.
+    let cfg = gen_cfg(gen_huge_string(WS_HTTP_HEADER_MAX_SIZE));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Use a huge value which the node will always reject.
+    let cfg = gen_cfg(gen_huge_string(WS_HTTP_HEADER_INVALID_SIZE));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Send an empty field.
+    let cfg = gen_cfg(String::new());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
```

Comment:
it would be more visible if we use `assert_eq(false, ...)` that we expect these tests to fail
small `!` is just a tiny character here 🤔

Reply:
Done - it looks more readable, but kinda strange -> let's keep it for the code readability reasons